src/QMCDrivers/Optimizers/DescentEngine.cpp

```diff
@@ -422,7 +422,9 @@ void DescentEngine::mpi_unbiased_ratio_of_means(int numSamples,
     y[5] += weight * n * d;
   }
 
-  my_comm_->allreduce(y);
+  //No need to reduce if there is only one MPI process
+  if(my_comm_->size() > 1)
+      my_comm_->allreduce(y);
```

Comment:
Without the protection, does the code run into problems? It is supposed to work unless my_comm was not set up properly.